### PR TITLE
Netcore fixes.

### PR DESCRIPTION
**Pull request by twitchax**
This change will cause the build to fail when a cmdlet name cannot be mapped to a documentation group.

In addition, there are some fixes to the Netcore build.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [x] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [x] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [x] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.
- [x] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [x] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [x] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [x] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [x] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [x] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by maddieclayton** [comment]
On demand run here: https://azuresdkci.westus2.cloudapp.azure.com/view/PowerShell/job/powershell-demand/224/

**Comment by twitchax** [comment]
@cormacpayne, can you take a look at this?  A while back, I added a property to the `build.proj` to stop on errors, but the script was not actually throwing and error.  This fixes that.

**Comment by maddieclayton** [comment]
@twitchax ping me when you're ready for me to review it

**Comment by twitchax** [comment]
@maddieclayton, probably good to go after build.

**Comment by twitchax** [comment]
@maddieclayton, ok!  I manually fixed for the pnm1: no reason not to, in my opinion.  The rest of the psd1 has stuff that depends on the build placing the files in the right places.

File: src/ResourceManager/Websites/AzureRM.Websites.Netcore.psd1 [hunk 1]
```diff
@@ -9,7 +9,7 @@
 @{
 
 # Script module or binary module file associated with this manifest.
-# RootModule = ''
+RootModule = '.\AzureRM.Websites.Netcore.psm1'
```
Comment (maddieclayton) [inline src/ResourceManager/Websites/AzureRM.Websites.Netcore.psd1]: I don't think this is the way we want to do it - when I asked Mark he said this would make debugging more difficult.  I'll try to figure out a better way to do this in the build.proj file

Comment (twitchax) [reply src/ResourceManager/Websites/AzureRM.Websites.Netcore.psd1]: @maddieclayton and @markcowl, what are the debug issues?  Maybe it's Netcore, but I can debug just fine the way it is.  What problems does this cause on desktop?

Also, this makes dynamic completers work without having to do a bunch of changes/fixes to UpdateModules and PublishModules.

Comment (maddieclayton) [reply src/ResourceManager/Websites/AzureRM.Websites.Netcore.psd1]: @twitchax see my new PR - this should fix the issue (I also added you as a reviewer)

File: tools/UpdateModules.ps1 [hunk 2]
```diff
@@ -265,31 +265,34 @@ if ($Profile -eq "Stack")
 }
 
 $resourceManagerRootFolder = "$packageFolder\$buildConfig\ResourceManager\AzureResourceManager"
-$publishToLocal = test-path $repositoryLocation
```
Comment (maddieclayton) [inline tools/UpdateModules.ps1]: Why did you delete this?

File: src/ResourceManager/Compute/AzureRM.Compute.Netcore.psd1 [hunk 3]
```diff
@@ -9,7 +9,7 @@
 @{
 
 # Script module or binary module file associated with this manifest.
-# RootModule = ''
+RootModule = '.\AzureRM.Compute.Netcore.psm1'
```
[the remark this answers is not in the document]
Comment (twitchax) [reply src/ResourceManager/Compute/AzureRM.Compute.Netcore.psd1]: Kept this for now based on discussion today.

File: tools/UpdateModules.ps1 (hunk 2 above)
Comment (twitchax) [reply tools/UpdateModules.ps1]: No action based on discussion today.

File: src/ResourceManager/Compute/Commands.Compute/Commands.Compute.Netcore.csproj [hunk 4]
```diff
@@ -182,4 +183,8 @@
     <Folder Include="VhdManagement\Model\Persistance\" />
   </ItemGroup>
 
+  <ItemGroup>
+    <Content Include="help\**\*" CopyToPublishDirectory="PreserveNewest" />
```
[the remark this answers is not in the document]
Comment (twitchax) [reply src/ResourceManager/Compute/Commands.Compute/Commands.Compute.Netcore.csproj]: Genius!  My bad.

**Comment by cormacpayne** [comment]
@azuresdkci test this please

File: src/ResourceManager/Storage/AzureRM.Storage.Netcore.psd1 [hunk 5]
```diff
@@ -54,11 +54,11 @@ PowerShellVersion = '5.1'
 RequiredModules = @(@{ModuleName = 'AzureRM.Profile.Netcore'; ModuleVersion = '0.10.0'; })
 
 # Assemblies that must be loaded prior to importing this module
-RequiredAssemblies = 'netcoreapp2.0\Microsoft.Azure.Management.Storage.dll',
-    'netcoreapp2.0\Microsoft.Data.Edm.dll',
-    'netcoreapp2.0\Microsoft.Data.OData.dll',
-    'netcoreapp2.0\Microsoft.WindowsAzure.Storage.dll',
-    'netcoreapp2.0\System.Spatial.dll'
+RequiredAssemblies = '.\Microsoft.Azure.Management.Storage.dll',
+    '.\Microsoft.Data.Edm.dll',
```
Comment (hovsepm) [inline src/ResourceManager/Storage/AzureRM.Storage.Netcore.psd1]: FYI, ".\" paths were not working in Linux in one of the beta builds. Please make sure this will note break Linux modules.

Comment (twitchax) [reply src/ResourceManager/Storage/AzureRM.Storage.Netcore.psd1]: This was fixed, and it works perfectly fine in linux. :)

**Comment by twitchax** [comment]
@cormacpayne, this should be ready. :)

**Comment by twitchax** [comment]
@cormacpayne, ping. :)
